sdk/core/azure-core-amqp/CHANGELOG.md

```diff
@@ -6,11 +6,12 @@
 
 ### Breaking Changes
 
+- Refactored AMQP headers to isolate internal-only types to "internal" subdirectory in headers.
+
```

Comment:
Are there any types intended for "public" use by SDK customers impacted by this change? Only then would it warrant mentioning as a breaking change. Otherwise, by convention, this refactor of moving internal-only types to different header locations isn't an end-customer breaking change (similar to any moves to _detail or actual private implementation details).

Reply:
If a customer had a line, "`#include <azure/core/amqp/connection.hpp>`", even though you did not use any definitions from it, it is a breaking change, your build will be broken until you remove that line.

Reply:
What Anton said.

Specifically, we know it's a breaking change because I needed to change the EventHubs code to reflect the change.

Also it's important to remember remember that "internal" types ARE public types. They are intended for consumption by the packages within the Azure SDK for C++. But they are allowed to change their public API contract within reason.